src/tests/cptests/Makefile

```diff
@@ -14,13 +14,13 @@ build-tests: cptests
 run-tests: cptests
 	./cptests
 
-cptests: cptests.o $(parent_objs) $(parent_test_objs)
-	$(CXX) -o cptests cptests.o $(parent_test_objects) $(parent_objs) $(ALL_TEST_LDFLAGS)
+# Use objects from tests/ (../) directory because we already have them
+cptests_objs = $(foreach obj,$(parent_objs),../$(obj))
```

Comment:
Creating a new variable, and using `foreach`, seems like overkill, just use a substitution reference. `$(parent_objs:%=../%)` should work and no need for a new `cptests_objs` variable.